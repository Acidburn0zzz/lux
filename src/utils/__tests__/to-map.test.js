/* @flow */

import toMap from '../to-map'

const keyer = ({ id }) => id

test('it can collect an array of items into a map', () => {
  const items = [
    { id: 1, name: 'Test 1' },
    { id: 2, name: 'Test 2' },
    { id: 3, name: 'Test 3' },
    { id: 4, name: 'Test 4' },
  ]

  expect(toMap(items, keyer)).toMatchSnapshot()
})

test('it ignores the item when the key is null or undefined', () => {
  const items = [
    { id: 1, name: 'Test 1' },
    { id: 2, name: 'Test 2' },
    { id: null, name: 'Test 3' },
    { id: undefined, name: 'Test 4' },
  ]

  expect(toMap(items, keyer)).toMatchSnapshot()
})
