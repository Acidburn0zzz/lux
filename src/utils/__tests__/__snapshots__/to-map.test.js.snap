// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it can collect an array of items into a map 1`] = `
Map {
  1 => Object {
    "id": 1,
    "name": "Test 1",
  },
  2 => Object {
    "id": 2,
    "name": "Test 2",
  },
  3 => Object {
    "id": 3,
    "name": "Test 3",
  },
  4 => Object {
    "id": 4,
    "name": "Test 4",
  },
}
`;

exports[`it ignores the item when the key is null or undefined 1`] = `
Map {
  1 => Object {
    "id": 1,
    "name": "Test 1",
  },
  2 => Object {
    "id": 2,
    "name": "Test 2",
  },
}
`;
